timvt/layer.py

```diff
@@ -147,38 +143,61 @@ async def get_tile(
                                 :xmax,
                                 :ymax,
                                 -- If EPSG is null we set it to 0
-                                {tms_epsg or 0}
+                                coalesce(:tms_srid, 0)
                             ),
                             :seg_size
                         ) AS geom
                 ),
+                bounds_geomcrs AS (
+                    SELECT
+                        CASE WHEN coalesce(:tms_srid, 0) != 0 THEN
+                            ST_Transform(bounds_tmscrs.geom, :geometry_srid)
+                        ELSE
+                            ST_Transform(bounds_tmscrs.geom, :tms_proj, :geometry_srid)
+                        END as geom
+                    FROM bounds_tmscrs
+                ),
                 mvtgeom AS (
                     SELECT ST_AsMVTGeom(
-                        ST_Transform(t.{geometry_column}, {out_srid}),
-                        bounds.geom,
+                        CASE WHEN :tms_srid IS NOT NULL THEN
+                            ST_Transform(t.:geometry_column, :tms_srid)
+                        ELSE
+                            ST_Transform(t.:geometry_column, :tms_proj)
+                        END,
+                        bounds_tmscrs.geom,
                         :tile_resolution,
                         :tile_buffer
-                    ) AS geom, {colstring}
-                    FROM {self.id} t, bounds
+                    ) AS geom, :fields
+                    FROM :tablename t, bounds_tmscrs, bounds_geomcrs
                     -- Find where geometries intersect with input Tile
                     -- Intersects test is made in table geometry's CRS (e.g WGS84)
                     WHERE ST_Intersects(
-                        t.{geometry_column}, {st_trans_expr}
-                    ) {limitstr}
+                        t.:geometry_column, bounds_geomcrs.geom
+                    ) LIMIT :limit
                 )
                 SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom
             """
 
-            return await conn.fetchval_b(
+            q, p = render(
                 sql_query,
+                tablename=V(self.id),
+                geometry_column=V(geometry_column),
+                fields=select_fields(*cols),
                 xmin=bbox.left,
                 ymin=bbox.bottom,
                 xmax=bbox.right,
                 ymax=bbox.top,
+                geometry_srid=funcs.cast(geometry_srid, "int"),
```

Comment:
why do we need casting here, should the table srid should always be an integer (https://github.com/developmentseed/timvt/pull/67/files#diff-bcf560ced05c5ff6a0fbd6081a51721df19813aaf75ccb0186d1b5aca097c5ccR87)